### PR TITLE
additional filters for wikidata

### PR DESCRIPTION
**Pull request by dibyaaaaax**
filters out wikidata items associated with pages in Wikipedia Category, Template, and Portal (eg: [Q3180](https://www.wikidata.org/wiki/Q3180))

### CODE REVIEW

File: mwtext/content_transformers/wikidata2words.py [hunk 1]
```diff
@@ -31,13 +31,21 @@ def transform(self, content):
...
     def get_claim_pid_index(self, claims_tuple):
         pid = claims_tuple[0]
         return self.pid_order_map.get(pid, len(self.pid_order_map))
 
+    def instanceof_nonzero_namespace(self, claims_tuples):
+        namespace_QIDs =  ['Q4167836', 'Q11266439', 'Q4663903']
+        return any((claims_tuple[0] == 'P31' and 
+                    claims_tuple[1] in namespace_QIDs)
+                   for claims_tuple in claims_tuples)
+
```
Comment (halfak) [inline mwtext/content_transformers/wikidata2words.py:48]: What's going on here?  It seems the function is checking for a namespace, but then it is matching against particular Qids?

Comment (dibyaaaaax) [reply mwtext/content_transformers/wikidata2words.py:48]: most of the Wikidata items that represent the pages with namespaces we are not interested in are "instance of" those QIDs. 
Another option I discussed with Isaac was to check all the wikis in the sitelink to see if each of them had a prefix of form "\<namespace name>\:".

Comment (galtay) [reply mwtext/content_transformers/wikidata2words.py:48]: in the past, I've had success with filtering any instances or subclasses of this https://www.wikidata.org/wiki/Q17442446

Comment (galtay) [reply mwtext/content_transformers/wikidata2words.py:48]: but in order to do that you have to have a list of all the subclasses .... the wikidata query service typically times out on a query like that .... but maybe you have some other means.  I parse all the subclass edges out of a wikidata dump as a preprocessing step (every statement that has P279), but its not super fast.

Comment (dibyaaaaax) [reply mwtext/content_transformers/wikidata2words.py:48]: @galtay you mean filtering out any item that is `instance of` or `subclass of` Q17442446 and its subclasses?

Comment (galtay) [reply mwtext/content_transformers/wikidata2words.py:48]: yes, that's the idea.  i'm not sure exactly what your filtering criteria are, but its one way to do these sorts of things.  you can see the subclasses in graph form here --  https://angryloki.github.io/wikidata-graph-builder/?property=P279&item=Q17442446&mode=reverse

Comment (halfak) [reply mwtext/content_transformers/wikidata2words.py:48]: I had a quick chat with @geohci about this.  I think that the right approach is to write separate script that gathers all of the relevant Qids and puts them into a data file and then we can load that data file as part of the filter method.  

This is a similar pattern to the one we used for extracting idioms for use in revscoring.  See https://github.com/wikimedia/revscoring/blob/master/revscoring/utilities/fetch_idioms.py (this script gathers idioms from wiktionary), https://github.com/wikimedia/revscoring/blob/master/revscoring/assets/enwiktionary_idioms.txt (here is the data file with the idioms), and https://github.com/wikimedia/revscoring/blob/master/revscoring/languages/english.py#L278 (here is where the data file is loaded and used).

**Comment by dibyaaaaax** [comment]
this PR has been updated in #19

**Pull request closed without merging**